sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -172,6 +172,11 @@ object ColumnPruning extends Rule[LogicalPlan] {
 
     case Project(projectList, Limit(exp, child)) =>
       Limit(exp, Project(projectList, child))
+
+    // push down project if possible when the child is sort
+    case p @ Project(projectList, s @ Sort(order, _, grandChild))
+      if (s.references -- p.outputSet).isEmpty =>
```

Comment:
You don't need `order` here, and with that removed you might be able to put the `if` on the same line.  Otherwise I think the if should be indented 4 characters.

Also, consider `s.references.subsetOf(p.outputSet)`, which more clearly says "the `Sort` can be evaluated with `output` of the `Project`" to me.